lib/storage/release.go

```diff
@@ -0,0 +1,255 @@
+}
+
+// GetStatus returns the release status.
+func (r *ReleaseV1) GetStatus() string {
+	return r.Status.Status
+}
+
+// GetRevision returns the release revision number.
+func (r *ReleaseV1) GetRevision() int {
+	return r.Status.Revision
+}
+
+// GetUpdated returns the release last update timestamp.
+func (r *ReleaseV1) GetUpdated() time.Time {
+	return r.Status.Updated
+}
+
+// GetLocator returns locator of the corresponding application package.
+func (r *ReleaseV1) GetLocator() loc.Locator {
+	return loc.Locator{
+		Repository: defaults.SystemAccountOrg,
+		Name:       r.Spec.ChartName,
+		Version:    r.Spec.ChartVersion,
+	}
+}
+
+// GetName returns the resource name.
+func (r *ReleaseV1) GetName() string {
+	return r.Metadata.Name
+}
+
+// SetName sets the resource name.
+func (r *ReleaseV1) SetName(name string) {
+	r.Metadata.Name = name
+}
+
+// GetMetadata returns the resource metadata.
+func (r *ReleaseV1) GetMetadata() services.Metadata {
+	return r.Metadata
+}
+
+// SetExpiry sets the resource expiration time.
+func (r *ReleaseV1) SetExpiry(expires time.Time) {
+	r.Metadata.SetExpiry(expires)
+}
+
+// Expires returns the resource expiration time.
+func (r *ReleaseV1) Expiry() time.Time {
+	return r.Metadata.Expiry()
+}
+
+// SetTTL sets the resource TTL.
+func (r *ReleaseV1) SetTTL(clock clockwork.Clock, ttl time.Duration) {
+	r.Metadata.SetTTL(clock, ttl)
+}
+
+// MarshalRelease marshals provided release resource to JSON.
+func MarshalRelease(release Release, opts ...services.MarshalOption) ([]byte, error) {
+	return json.Marshal(release)
+}
+
+// UnmarshalRelease unmarshals release resource from the provided JSON data.
+func UnmarshalRelease(data []byte) (Release, error) {
+	jsonData, err := utils.ToJSON(data)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var header services.ResourceHeader
+	err = json.Unmarshal(jsonData, &header)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	switch header.Version {
+	case services.V1:
+		var release ReleaseV1
+		err := utils.UnmarshalWithSchema(GetReleaseSchema(), &release, jsonData)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &release, nil
+	}
+	return nil, trace.BadParameter("%v resource version %q is not supported",
+		KindRelease, header.Version)
+}
+
+// GetReleaseSchema returns the full release resource schema.
+func GetReleaseSchema() string {
+	return fmt.Sprintf(services.V2SchemaTemplate, services.MetadataSchema,
+		ReleaseV1Schema, "")
+}
+
+// ReleaseV1Schema defines the release resource schema.
+var ReleaseV1Schema = fmt.Sprintf(`{
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "chart_name": {"type": "string"},
+    "chart_version": {"type": "string"},
+    "app_version": {"type": "string"},
+    "namespace": {"type": "string"}
+  }
+},
+"status": {
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "status": {"type": "string"},
+    "revision": {"type": "number"},
```

Comment:
Can this be float?

Reply:
No, why? Can `number` be a float? We use it everywhere AFAICS.